### PR TITLE
D8CORE-1370: Alter menu form and status messages for config_readonly users

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW 

# Summary
- Removes warning label and fields from form page where some items were editable and others were not.

# Review By (Date)
- Feb 26th

# Urgency
- Med/Low

# Steps to Test

1. Do a full site build with stanford_profile
2. Check out this branch and clear cache
3. Set `$settings['config_readonly'] = TRUE;` in your `local.settings.php` file.
4. Log in as a site_manager and edit the form at `/admin/structure/menu/manage/main`
5. Log in as an administrator and edit the form at `/admin/structure/menu/manage/main`

# Affected Projects or Products
- D8CORE

# Associated Issues and/or People
- D8CORE-1370

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
https://www.drupal.org/project/config_readonly/issues/2892631

**Comment by pookmish** [comment]
is this still needed with https://github.com/SU-SWS/acsf-cardinalsites/pull/88 ?

**Comment by sherakama** [comment]
Yes, even though the form is now open to re-arrange menu items we still want to hide the options that cannot be changed and remove the warning message.

**Comment by pookmish** [comment]
Makes sense to me :+1:

File: stanford_profile.profile [hunk 1]
```diff
@@ -167,6 +168,28 @@ function stanford_profile_node_access(NodeInterface $node, $op, AccountInterface
...
+  if (!$read_only) {
+    return;
+  }
+
+  // If the form is locked, hide the config you cannot change from users without
+  // the know how.
+  $access = \Drupal::currentUser()->hasPermission('Administer menus and menu items');
+  $form['label']['#access'] = $access;
+  $form['description']['#access'] = $access;
+  $form['id']['#access'] = $access;
```
Comment (pookmish) [inline stanford_profile.profile]: Not really needed but :man_shrugging:  :neutral_face:

Comment (sherakama) [reply stanford_profile.profile]: It did show up when the label was hidden. I thought it was extra at first too.

**Pull request merged**